pinot-core/src/main/java/org/apache/pinot/core/common/DataTableFactory.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.common;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.common.utils.DataTable;
+import org.apache.pinot.core.common.datatable.DataTableBuilderV2V3;
+import org.apache.pinot.core.common.datatable.DataTableBuilderV4;
+import org.apache.pinot.core.common.datatable.DataTableImplV2;
+import org.apache.pinot.core.common.datatable.DataTableImplV3;
+import org.apache.pinot.core.common.datatable.DataTableImplV4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class DataTableFactory {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DataTableFactory.class);
+  public static final int VERSION_2 = 2;
+  public static final int VERSION_3 = 3;
+  public static final int VERSION_4 = 4;
+  public static final int DEFAULT_VERSION = VERSION_3;
+
+  private static int _version = DataTableFactory.DEFAULT_VERSION;
```

Comment:
put `private static` all together. Also make the constant name upper cases?

Reply:
this field is not final. thus individually separated